### ISSUE TITLE
diff column sorting also sorts on the string values, not on the diff %

### ISSUE THREAD

**Issue by Rperry2174** [grafana/pyroscope#2850]
### Bug description

diff column sorting is not working as expected for me: it also sorts on the string values, not only on the diff %. This makes it slightly harder to see where the biggest impact is.

https://github.com/grafana/pyroscope-app-plugin/assets/35603/d2cdb410-4aef-485a-9703-1034dd36e66f

cc @aocenas

**Comment by aocenas** [comment]
It doesn't sort on the string value. It treats `new` as +Infinity and `removed` as `-infinity`. The issue here seems to be the NaN values which are uncomparable. I am not sure what kind of data this is as it's 0 on both sides so I would assume those items shouldn't actualy be there. Do you know why such items are in the flame graph or what should be the diff of such values if they are supposed to be there?

**Comment by Rperry2174** [comment]
Yeah that makes sense that those values are there and I see now why that's an issue. 

I'd expect sorting in descending order to go:

- +infinity
- +100% 
- 0%
- -100%
- -infinity 

And ascending to be the opposite
